mlir/lib/Conversion/TosaToRock/TosaToRock.cpp

```diff
@@ -215,6 +215,58 @@ makeRockConv(ConversionPatternRewriter &rw, Operation *op, Value input,
   return cop;
 }
 
+static bool isTosaReduction(Operation *op) {
+  return isa<tosa::ReduceMaxOp, tosa::ReduceSumOp, tosa::ReduceMinOp,
+             tosa::ReduceProdOp, tosa::ReduceAllOp, tosa::ReduceAnyOp>(op);
+}
+
+static Value traceToRes(Value tensor, DenseMap<Value, Value> &cache,
+                        Value expectedTensor) {
+  auto cached = cache.find(tensor);
+  if (cached != cache.end())
+    return cached->second;
```

Comment:
use cache.contains(..)

Reply:
happy to use contains but I think it's less efficient as we query the cache twice.

Reply:
hmm.. its constant time access. just felt cleaner.
I ll let you decide.